mithril-aggregator/src/artifact_builder/cardano_database.rs

```diff
@@ -0,0 +1,213 @@
+        let mut directory_size = 0;
+        for entry in entries {
+            let path = entry
+                .with_context(|| format!("Failed to read directory entry in {:?}", path))?
+                .path();
+            directory_size += compute_fs_entry_size(&path)?;
+        }
+
+        return Ok(directory_size);
+    }
+
+    Ok(0)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use mithril_common::{
+        digesters::DummyImmutablesDbBuilder,
+        entities::{ProtocolMessage, ProtocolMessagePartKey},
+        test_utils::{fake_data, TempDir},
+    };
+
+    use super::*;
+
+    fn get_test_directory(dir_name: &str) -> PathBuf {
+        TempDir::create("cardano_database", dir_name)
+    }
+
+    #[test]
+    fn should_compute_the_size_of_the_uncompressed_database_only_immutable_ledger_and_volatile() {
+        let test_dir = get_test_directory("should_compute_the_size_of_the_uncompressed_database_only_immutable_ledger_and_volatile");
+
+        let immutable_file_size = 777;
+        let ledger_file_size = 6666;
+        let volatile_file_size = 99;
+        DummyImmutablesDbBuilder::new(test_dir.as_os_str().to_str().unwrap())
+            .with_immutables(&[1, 2])
+            .set_immutable_file_size(immutable_file_size)
+            .with_ledger_files(vec!["blocks-0.dat".to_string()])
+            .set_ledger_file_size(ledger_file_size)
+            .with_volatile_files(vec!["437".to_string(), "537".to_string()])
+            .set_volatile_file_size(volatile_file_size)
+            .build();
+        // Number of immutable files = 2 × 3 ('chunk', 'primary' and 'secondary').
+        let expected_total_size =
+            (2 * 3 * immutable_file_size) + ledger_file_size + (2 * volatile_file_size);
+
+        std::fs::write(test_dir.join("non_computed_file.txt"), "file inside root").unwrap();
+        let non_computed_dir = test_dir.join("non_computed_dir");
+        std::fs::create_dir(&non_computed_dir).unwrap();
+        std::fs::write(
+            non_computed_dir.join("another_non_computed_file.txt"),
+            "file inside a non computed directory",
+        )
+        .unwrap();
+
+        let total_size = compute_uncompressed_database_size(&test_dir).unwrap();
+
+        assert_eq!(expected_total_size, total_size);
+    }
+
+    #[tokio::test]
+    async fn should_compute_valid_artifact() {
+        let test_dir = get_test_directory("should_compute_valid_artifact");
+
+        DummyImmutablesDbBuilder::new(test_dir.as_os_str().to_str().unwrap())
+            .with_immutables(&[1])
+            .set_immutable_file_size(100)
+            .with_ledger_files(vec!["blocks-0.dat".to_string()])
+            .set_ledger_file_size(100)
+            .with_volatile_files(vec!["437".to_string()])
+            .set_volatile_file_size(100)
+            .build();
+        // Number of immutable files = 1 × 3 ('chunk', 'primary' and 'secondary').
+        let expected_total_size = (3 * 100) + 100 + 100;
```

Comment:
Same remark as above on readability of the test

Reply:
The test was simplified based on the different comments on the PR: https://github.com/input-output-hk/mithril/pull/2159/files#diff-aab140d744a307dd040e3dba02956b21858cb54510ea96a58ab094124897da8eR139-R150